test/tbb/test_parallel_for.cpp

```diff
@@ -462,3 +462,60 @@ TEST_CASE("parallel_for constraints") {
 #if _MSC_VER
 #pragma warning (pop)
 #endif
+
+// Define a simple functor to use with parallel_for
+struct SimpleFunctor {
+    void operator()(const tbb::blocked_range<size_t>& r) const {
+        for (size_t i = r.begin(); i != r.end(); ++i) {
+            // Perform some operation on the range elements
+            // For simplicity, we'll just ensure each element is visited
+        }
+    }
+};
+
+void TestNumaPartitionerSimple() {
+    const size_t N = 1000;
+    std::vector<int> vec(N, 1);
+    
+    tbb::blocked_range<size_t> range(0, N);
+    SimpleFunctor functor;
+    tbb::affinity_partitioner ap;
+    tbb::numa_partitioner<tbb::affinity_partitioner> n_partitioner(ap);
+
+    // Test parallel_for with numa_partitioner
+    parallel_for(range, functor, n_partitioner);
+
+    // Verify results (for now, just check if the function runs without errors)
+    CHECK(true);
+}
+
+void TestNumaPartitionerWithBody() {
+    const size_t N = 1000;
+    std::vector<int> vec(N, 0);
+    
+    tbb::blocked_range<size_t> range(0, N);
+    
+    auto body = [&](const tbb::blocked_range<size_t>& r) {
+        for (size_t i = r.begin(); i != r.end(); ++i) {
+            vec[i] = 1; // Set each element to 1
+        }
+    };
+
+    tbb::affinity_partitioner ap;
+    tbb::numa_partitioner<tbb::affinity_partitioner> n_partitioner(ap);
+    
+    // Test parallel_for with numa_partitioner and a lambda body
+    parallel_for(range, body, n_partitioner);
```

Comment:
You need a test with more than one parallel_for invocation. I think that would uncover some of the design issues.